src/machine/machine_state_impl.rs

```diff
@@ -514,6 +515,14 @@ impl MachineState {
                                         return Some(n1.cmp(&n2));
                                     }
                                 }
+			        (HeapCellValueTag::Cons, ptr) => {
```

Comment:
This is unsafe. After verifying that `ptr` is a stream as below, `Some(n1)` should be compared to `stream.options().get_alias()` (without the unwrap). The rest is fine.